google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/RemoveFeedItemAttributeValue.java

```diff
@@ -0,0 +1,257 @@
+    // Gets the index of the attribute value that will be removed.
+    int attributeIndex = getAttributeIndex(feedItem, feedItemAttributeValue);
+
+    // Returns the feed item with the removed FeedItemAttributeValue. Any FeedItemAttributeValues
+    // that are not included in the updated FeedItem will be removed from the FeedItem, which is
+    // why you must create the FeedItem from the existing FeedItem and set the field(s) that will
+    // be removed.
+    return FeedItem.newBuilder(feedItem).removeAttributeValues(attributeIndex).build();
+  }
+
+  /**
+   * Retrieves a feed item and its attribute values given a resource name.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID in which to create criterion.
+   * @param feedItemResourceName feed item resource name.
+   * @return FeedItem with the given resource name.
+   */
+  private FeedItem getFeedItem(
+      GoogleAdsClient googleAdsClient, long customerId, String feedItemResourceName) {
+    // Constructs the query.
+    String query =
+        "SELECT feed_item.attribute_values FROM feed_item WHERE feed_item.resource_name "
+            + "= '"
+            + feedItemResourceName
+            + "'";
+
+    // Constructs the request.
+    SearchGoogleAdsRequest request =
+        SearchGoogleAdsRequest.newBuilder()
+            .setCustomerId(String.valueOf(customerId))
+            .setPageSize(PAGE_SIZE)
+            .setQuery(query)
+            .build();
+
+    // Issues the search request.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      SearchPagedResponse searchPagedResponse = googleAdsServiceClient.search(request);
+      // Returns the feed item attribute values.
+      return searchPagedResponse.getPage().getResponse().getResults(0).getFeedItem();
+    }
+  }
+
+  /**
+   * Gets the ID of the flight price attribute. This is needed to specify which
+   * FeedItemAttributeValue will be updated in the given FeedItem.
+   *
+   * @param feedItem the FeedItem that will be updated.
+   * @param newFeedItemAttributeValue the new FeedItemAttributeValue that will be updated.
+   * @return Int the index of the attribute.
+   */
+  private int getAttributeIndex(
+      FeedItem feedItem, FeedItemAttributeValue newFeedItemAttributeValue) {
+    int attributeIndex = 0;
+
+    // Loops through attribute values to find the index of the FeedItemAttributeValue to update
+    for (FeedItemAttributeValue feedItemAttributeValue : feedItem.getAttributeValuesList()) {
+      // Checks if the current feedItemAttributeValue is the one we are updating
+      if (feedItemAttributeValue.getFeedAttributeId().getValue()
+          == newFeedItemAttributeValue.getFeedAttributeId().getValue()) {
+        break;
+      }
+      attributeIndex++;
+    }
+
+    return attributeIndex;
```

Comment:
This doesn't handle the case where no attribute is found. It looks like you want to throw RuntimeException in those cases.